apps/src/studio/studio.js

```diff
@@ -1546,6 +1546,15 @@ Studio.execute = function() {
   Studio.waitingForReport = false;
   Studio.response = null;
 
+  var usesFunctionalBlocks = level.useContractEditor;
```

Comment:
Hmm, I'm pretty sure it's theoretically possible to use functional blocks without the contract editor, but it may be the case that something breaks when that's not the case.

One alternative might be to make `hasUnfilledBlock` become `hasUnfilledFunctionalBlock`